api/src/controllers/young/inscription.js

```diff
@@ -45,4 +45,138 @@ router.put("/profile", passport.authenticate("young", { session: false, failWith
+        otherwise: Joi.isError(new Error()),
+      }),
+      grade: Joi.alternatives().conditional("schooled", {
+        is: "true",
+        then: Joi.string().trim().required().valid("3eme", "2nd", "1ere", "1ere CAP", "Terminale", "Terminale CAP", "SEGPA", "Classe relais", "Autre"),
+        otherwise: Joi.isError(new Error()),
+      }),
+
+      employed: Joi.alternatives().conditional("schooled", { is: "false", then: Joi.string().trim().required().valid("true", "false"), otherwise: Joi.string().default("false") }),
+
+      //Field validation for situation depending on schooled and employed
+      situation: Joi.alternatives().conditional("schooled", {
+        is: "true",
+        then: Joi.string()
+          .trim()
+          .required()
+          .valid(
+            YOUNG_SITUATIONS.GENERAL_SCHOOL,
+            YOUNG_SITUATIONS.PROFESSIONAL_SCHOOL,
+            YOUNG_SITUATIONS.AGRICULTURAL_SCHOOL,
+            YOUNG_SITUATIONS.SPECIALIZED_SCHOOL,
+            YOUNG_SITUATIONS.APPRENTICESHIP,
+          ),
+        otherwise: Joi.alternatives().conditional("employed", {
+          is: "true",
+          then: Joi.string().trim().required().valid(YOUNG_SITUATIONS.EMPLOYEE, YOUNG_SITUATIONS.INDEPENDANT, YOUNG_SITUATIONS.SELF_EMPLOYED, YOUNG_SITUATIONS.ADAPTED_COMPANY),
+          otherwise: Joi.string().trim().required().valid(YOUNG_SITUATIONS.POLE_EMPLOI, YOUNG_SITUATIONS.MISSION_LOCALE, YOUNG_SITUATIONS.CAP_EMPLOI, YOUNG_SITUATIONS.NOTHING),
+        }),
+      }),
+    }).validate(req.body);
+
+    if (error) {
+      return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS });
+    }
+
+    if (value.livesInFrance === "true") {
+      value.hostLastName = "";
+      value.hostFirstName = "";
+      value.hostRelationship = "";
+      value.foreignCountry = "";
+      value.foreignCity = "";
+      value.foreignZip = "";
+      value.foreignAddress = "";
+    }
+
+    if (value.schooled === "false") {
+      value.schoolName = "";
+      value.schoolId = "";
+      value.schoolCountry = "";
+      value.schoolCity = "";
+      value.schoolDepartment = "";
+      value.grade = "";
+    }
+
+    delete value.livesInFrance;
+    value.inscriptionStep = STEPS.AVAILABILITY;
+
+    const young = await YoungObject.findById(req.user._id);
+    if (!young) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
+
+    if (!canUpdateYoungStatus({ body: value, current: young })) return res.status(403).send({ ok: false, code: ERRORS.OPERATION_UNAUTHORIZED });
+
+    young.set(value);
+    await young.save({ fromUser: req.user });
+
+    inscriptionCheck(value, young, req);
+
+    return res.status(200).send({ ok: true, data: young });
+  } catch (error) {
+    capture(error);
+    return res.status(500).send({ ok: false, code: ERRORS.SERVER_ERROR, error });
```

Comment:
```suggestion
    return res.status(500).send({ ok: false, code: ERRORS.SERVER_ERROR });
```

Dans une autre tache, on ne fait plus suivre les erreurs côté client quand elles sont innatendues : elles donnent potentiellement des informations